backend/controllers/fieldController.js

```diff
@@ -0,0 +1,126 @@
+const Field = require('../models/FieldSchema');
+const { validationResult } = require('express-validator');
+const asyncHandler = require('express-async-handler');
+const Admin = require("../models/adminSchema");
+
+// @desc Get all fields
+// @route GET /api/fields
+// @access Private -> Admin only
+const getFields = asyncHandler(async (req, res) => {
+  const admin = await Admin.findById(req.admin.id);
+  if (!admin) {
+    res.status(401);
+    throw new Error("Admin not found");
+  }
+  res.send(await Field.find({}));
+});
+
+// @desc Get field by name
+// @route GET /api/fields/:name
+// @access Private -> Admin only
+const getFieldByName = asyncHandler(async (req, res) => {
+  const admin = await Admin.findById(req.admin.id);
+  if (!admin) {
+    res.status(401);
+    throw new Error("Admin not found");
+  }
+
+  constfieldName = req.params.name;
+  const field = await Field.findOne({ [fieldName]: { $exists: true } });
+  if (!field) {
+    return res.status(404).json({ message: "No field found" });
+  }
+
+  res.json({ [fieldName]: field[fieldName] });
+});
+
+// @desc Add item to field
+// @route POST /api/fields
```

Comment:
I messed this up on the issue can you change to 
```suggestion
// @route PUT /api/fields
```